src/Microsoft.DocAsCode.Build.RestApi/Swagger/Internals/SwaggerJsonBuilder.cs

```diff
@@ -19,16 +22,25 @@ internal class SwaggerJsonBuilder
         private const string InternalRefNameKey = "x-internal-ref-name";
         private const string InternalLoopRefNameKey = "x-internal-loop-ref-name";
 
-        public SwaggerObjectBase Read(JsonReader reader)
+        public SwaggerObjectBase Read(string swaggerPath)
         {
-            _documentObjectCache = new Dictionary<string, SwaggerObjectBase>();
-            var token = JToken.ReadFrom(reader);
-            var swagger = Build(token);
-            RemoveReferenceDefinitions((SwaggerObject)swagger);
-            return ResolveReferences(swagger, new Stack<string>());
+            using (JsonReader reader = new JsonTextReader(File.OpenText(swaggerPath)))
+            {
+                _documentObjectCache = new Dictionary<string, SwaggerObjectBase>();
+                var token = JToken.ReadFrom(reader);
+
+                var swaggerDir = Path.GetDirectoryName(swaggerPath);
+                if (string.IsNullOrEmpty(swaggerDir))
```

Comment:
in this method the path is full path, to open read in line 27

---
In reply to: [93162486](https://github.com/dotnet/docfx/pull/1019#discussion_r93162486) [](ancestors = 93162486)